src/import-export/bin/export-data

```diff
@@ -80,20 +83,78 @@ export_data()
 	fi
 }
 
+export_keys()
+{
+        backup="$1"
+	occ_err="/tmp/nextcloud_occ.err"
+	rsync_err="/tmp/nextcloud_rsync.err"
+	cat /dev/null > "$rsync_err"
+	
+        echo "Exporting encryption keys..."
+	encryption_status="$(nextcloud.occ encryption:status 2>$occ_err | awk 'NR==1{print $3}')"
+	if [ "$(cat $occ_err)" != "Nextcloud is in maintenance mode - no apps have been loaded" ]; then
+		# ignore the maintenance mode message but print out other errors.
+		echo "WARNING: nextcloud.occ reported an error checking the status of encryption:"
+		cat "$occ_err"
+	fi
+	if [ "$encryption_status" = "false" ]; then
+		# because these encryption directories do not exist until encryption is enabled,
+		# rsync will throw a bunch of directory not found errors
+		echo "Encryption does not appear to be enabled, skipping."
+	else
+		# back up system-wide master keys, public sharing keys, recovery keys
+		echo "Exporting system-wide keys..."
+		if ! rsync -ah --info=progress2 "$NEXTCLOUD_DATA_DIR"/files_encryption "${backup}"/keys/ 2>>"$rsync_err"; then
```

Comment:
I'm a smidge uncomfortable with this. Does this directory only ever contain keys?

Reply:
As far as I can tell, yes.  From the [official documentation](https://docs.nextcloud.com/server/16/admin_manual/configuration_files/encryption_configuration.html):

> The encryption keys are stored in the following directories:
> 
> `data/<user>/files_encryption`
>     Users’ private keys and all other keys necessary to decrypt the users’ files
> `data/files_encryption`
>     private keys and all other keys necessary to decrypt the files stored on a system wide external storage 

And even more detail is provided [here](https://github.com/syseleven/nextcloud-tools/blob/master/server-side-encryption.md#file-types) that is waiting to be merged into the official docs.